version: "3.8"

services:
  app:
    build:
      args:
        user: 1000
        uid: 1000
      context: ./
      dockerfile: Dockerfile
    image: calambre
    container_name: calambre-app
    restart: unless-stopped
    working_dir: /var/www/
    volumes:
      - ./calambre:/var/www
      - ./.docker/xdebug.ini:/usr/local/etc/php/conf.d/docker-php-ext-xdebug.ini
      - ./.docker/xdebug_error_reporting.ini:/usr/local/etc/php/conf.d/error_reporting.ini
    networks:
      - calambre
    extra_hosts:
      - "host.docker.internal:host-gateway"

  database:
    image: postgres
    container_name: calambre-db
    restart: unless-stopped
    user: 1000:1000
    environment:
      POSTGRES_PASSWORD: postgres
      POSTGRES_DB: calambre_db
    volumes:
      - ./.docker/postgres-data:/var/lib/postgresql/data
    ports:
      - '5432:5432'
    networks:
      - calambre

  nginx:
    image: nginx:alpine
    container_name: calambre-nginx
    restart: unless-stopped
    ports:
      - 8080:80
    volumes:
      - ./calambre/:/var/www
      - ./.docker/nginx:/etc/nginx/conf.d/
    networks:
      - calambre


  mailhog:
    image: mailhog/mailhog
    container_name: calambre-mail
    ports:
      - "1025:1025"
      - "8025:8025"
    networks:
      - calambre

networks:
  calambre:
    driver: bridge

